src/coreclr/tryrun.cmake

```diff
@@ -100,16 +103,21 @@ elseif(TARGET_ARCH_NAME MATCHES "^(armel|arm|arm64|x86)$" OR FREEBSD OR ILLUMOS)
 
 
   if(ALPINE_LINUX)
+    set_cache_value(HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP_EXITCODE 1)
```

Comment:
Why do we need these changes?

Reply:
This is a leftover from linking System.Native.  I reverted most of that to not be confusing until we get to System.Native, but missed a few things like this. 
It will be needed eventually.

Reply:
Can we include the config from the libraries? The config gets pretty complex, it would not be a good idea to duplicate it.

Reply:
I've essentially copied the relevant parts of config. I think I can just remove the duplicate part from libs' config.

Reply:
How is that going to work for Mono and NativeAOT? Does it mean that we will have to have the config duplicated in all those places?

Reply:
Ugh. We actually have two config files for coreclr and libs. (I assumed one uses another and adds few extras, since libs tryrun.cmake is so small)

What I did does not work for coreclr either. It is ok with Linux, which I can test locally, but not ok for OSX. 

If we have a global place for tryrun.cmake that everyone uses, this should go there.

Reply:
Alternatively, the relevant parts could be factored out of the libs' tryrun and included from both coreclr and libs.
I think I will do that.

It feels like we should not have separate "tryrun.cmake" files for different partitions. 
Maybe I do not understand completely what it does. It looks like it just pre-sets and caches a bunch of capability variables for a few known platforms. (the list of "known" platforms vary slightly between partitions, not sure if intentionally)

Reply:
Discussed the `tryrun.cmake` with @janvorli . There is no big reason why we have 3 of these files (one per partition), other than historically when we had different repos.

The work that is done there is prepopulating cached values in cross-compiling scenarios. That is a cheap operation so there should be no issues with unifying/merging these files into one under `eng`.

Reply:
I have moved the combined `tryrun.cmake` into `/eng` and deleted the others. No more duplication on this part.